api/kv.go

```diff
@@ -283,7 +286,36 @@ func (k *KV) deleteInternal(key string, params map[string]string, q *WriteOption
 
 // Txn is used to apply multiple KV operations in a single, atomic transaction.
 // Note that Go will perform the required base64 encoding on the values
-// automatically because the type is a byte slice.
+// automatically because the type is a byte slice. Transactions are defined as a
+// list of operations to perform, using the KVOp constants and KVTxnOp structure
+// to define operations. If any operation fails, none of the changes are applied
+// to the state store.
```

Comment:
Right now there's only a single write transaction at the state store level at any given time. If your transaction bounces and won't apply, there are a bunch of cases where you'd need to take specific action before you retry (update an index, for example). For the cases of locks, we already have retry helpers for single locks, so I'd like to avoid adding a retry thing here.